src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs

```diff
@@ -387,30 +387,37 @@ private void AddAnonymousTypeName(INamedTypeSymbol symbol)
 
         private void AddTupleTypeName(INamedTypeSymbol symbol)
         {
-            // TODO: revise to generate user-friendly name 
-            var members = string.Join(", ", symbol.GetMembers().OfType<IFieldSymbol>().Select(CreateTupleTypeMember));
+            var tupleType = (TupleTypeSymbol)symbol;
```

Comment:
Is it safe to do this cast? Could it be a VB symbol, for example (once we add support for tuples to VB)?